### ISSUE TITLE
Support for other plugins determining servers and virtual-host accounting.

### ISSUE THREAD

**Issue by HarvelsX** [kyngs/LibreLogin#96]
Hello! Unfortunately, the current version of the plugin ignores the presence of other plugins that determine the server destination for the player.
If you use the internal function `remember-last-server`, it does not take into account the virtual-host section (analog of force-host)

**Comment by kyngs** [comment]
> Hello! Unfortunately, the current version of the plugin ignores the presence of other plugins that determine the server destination for the player.

Yeah, this is problematic, sometimes this could be unwanted, sometimes not. Not really sure how to resolve this.

> If you use the internal function remember-last-server, it does not take into account the virtual-host section (analog of force-host)

What behavior do you suggest? When someone is joining from a forced host, should the plugin remember the last server from the host, or just ignore the last server completely?

**Comment by bivashy** [comment]
> > Hello! Unfortunately, the current version of the plugin ignores the presence of other plugins that determine the server destination for the player.
> 
> Yeah, this is problematic, sometimes this could be unwanted, sometimes not. Not really sure how to resolve this.
> 
> > If you use the internal function remember-last-server, it does not take into account the virtual-host section (analog of force-host)
> 
> What behavior do you suggest? When someone is joining from a forced host, should the plugin remember the last server from the host, or just ignore the last server completely?

What if we would have "whitelist-server". If some plugin will try to send player into another server, and it is included in whitelist, LibreLogin will let player join into server.

**Comment by HarvelsX** [comment]
> Yeah, this is problematic, sometimes this could be unwanted, sometimes not. Not really sure how to resolve this.

Be the very last one to make the determination and if the user needs to be rerouted to an authorization server, remember where he should have connected before and reroute to where you need to go with status tracking.

> What behavior do you suggest? When someone is joining from a forced host, should the plugin remember the last server from the host, or just ignore the last server completely?

A good approach would be to keep track of the last server relative to the connecting host, this is implemented in bungee and is quite well proven.

**Comment by kyngs** [comment]
I absolutely agree that determining on which server should the player land is absolutely out of the scope of an authorization plugin. However, people requested it, so I implemented it. The LobbyServerChooseEvent does allow you to change the server where the player will be headed to. 
As far as the forced hosts' issue goes, I would need to restructure the database before I could implement it.

**Comment by HarvelsX** [comment]
> The LobbyServerChooseEvent does allow you to change the server where the player will be headed to.

@kyngs, I believe it's easy to change this by listening to an event, but don't expect every plugin to use your API to do this. 
And why do that when it's easier to implement the destination mechanism described above in your plugin?

**Comment by kyngs** [comment]
that could work ig

**Comment by kyngs** [comment]
I'm refocusing this issue only on the forced hosts' problem. For the other problem, I'll track #12

**Comment by gamer50082** [comment]
forced host works
